### ISSUE TITLE
Question about hyperparameters

### ISSUE THREAD

**Issue by leo2105** [omarsayed7/Deep-Emotion#4]
Hi, thanks for the contribution
You have the training prepared for FER2013 right? According o the hyperparameters selected in the paper, they are wrong.. for example line 25 in deep_emotion.py. Shouldnt be `kernel_size=3` instead of 7? I dont know if there are other mistakes.
Another one, `test.py` doesnt have emotion column, so I had to use `icml_face_data.csv` instead to split for train, valid and test.
And last one, I want to know if you can pass me the hyperparameters for each dataset? Because before each modification I got at most 52% test accuracy for FER2013.
Greetings

**Comment by patilabhijit007** [comment]
Same here.
Epoch: 100      Training Loss: 0.00913613       Validation Loss 0.01061922      Training Acuuarcy 55.425%      Validation Acuuarcy 49.763%

**Comment by Payne214** [comment]
Me too

**Comment by shabaz2077** [comment]
How did u gyz solve the 
validation_dataset= Plain_Dataset(csv_file=validationcsv_file, img_dir = validation_img_dir, datatype = 'val', transform = transformation)
As it calls it put up an key error as it doesn't consists of 'emotion' column so how to solve this error

**Comment by shabaz2077** [comment]
> Hi, thanks for the contribution
> You have the training prepared for FER2013 right? According o the hyperparameters selected in the paper, they are wrong.. for example line 25 in deep_emotion.py. Shouldnt be `kernel_size=3` instead of 7? I dont know if there are other mistakes.
> Another one, `test.py` doesnt have emotion column, so I had to use `icml_face_data.csv` instead to split for train, valid and test.
> And last one, I want to know if you can pass me the hyperparameters for each dataset? Because before each modification I got at most 52% test accuracy for FER2013.
> Greetings

How did u use icml_face_data.csv rather than splitting

**Comment by shabaz2077** [comment]
> Same here.
> Epoch: 100 Training Loss: 0.00913613 Validation Loss 0.01061922 Training Acuuarcy 55.425% Validation Acuuarcy 49.763%

How did ur model just ran without an error bruh??

**Comment by rafalafel21** [comment]
For everyone asking how to get the model to run:
There is an error in the code which makes it so that final_test.csv and val.csv don't have labels, aka the emotion column. You have to download the FER2013 dataset from kaggle, convert that into a csv, then put it into pandas or something and remove all ROWS with label 'Training'. Then remove the COLUMN labeled 'Usage'. Convert your data frame with those rows and columns removed to a csv. Then make a generate_data object from the file generate_data.py from the repo initialized with the path to the csv you just made in pandas. Then call the 'split_test' method on said object. Then replace the final_test.csv and val.csv files that you initially got out of the repo with the filed returned form the split_test method INSIDE of the data repository you were initially using. Now your model should run without errors. Lol all that might be useless though because I have still only been able to achieve up to 52% testing accuracy with even 700 epochs? Let me know if anyone finds good hyper parameters.

**Comment by Muista** [comment]
> For everyone asking how to get the model to run:
> There is an error in the code which makes it so that final_test.csv and val.csv don't have labels, aka the emotion column. You have to download the FER2013 dataset from kaggle, convert that into a csv, then put it into pandas or something and remove all ROWS with label 'Training'. Then remove the COLUMN labeled 'Usage'. Convert your data frame with those rows and columns removed to a csv. Then make a generate_data object from the file generate_data.py from the repo initialized with the path to the csv you just made in pandas. Then call the 'split_test' method on said object. Then replace the final_test.csv and val.csv files that you initially got out of the repo with the filed returned form the split_test method INSIDE of the data repository you were initially using. Now your model should run without errors. Lol all that might be useless though because I have still only been able to achieve up to 52% testing accuracy with even 700 epochs? Let me know if anyone finds good hyper parameters.

I suppose that the model given in deep_emotion.py  may have some problems. I didn't use the train code in this project. I just copy the model file into my train project instead. And the result on JAFFE dataset is only about 50% val accuray which is largely less than the VGG16 version. I think the output of stn module should be Multiplied by the feature map after conv layer, not before the conv layer(as the picture in the paper, but I am not familiar with the stn module,so i am not sure). The attention module changes the input pic. I think it may be wrong.It should change the feature maps.

**Comment by omarsayed7** [comment]
> 55.425

https://github.com/omarsayed7/Deep-Emotion/commit/bccde6bc12d194af2dcdef579534806d81deeb86

**Comment by omarsayed7** [comment]
> icml_face_data

The paper didn't provide any hyperparameters or training trails, or tips, also this is not the official documentation of the paper, I tried to get that accuracy and tried to change many things including the loss function, network architecture, and even the hyperparameters.
